### DISCUSSION TITLE
ADC Frequency Limits for the 3226

### DISCUSSION THREAD

**Discussion by clazarowitz** [SpenceKonde/megaTinyCore#1052] [Q&A]
I'm not trying to exceed specs here, just trying to figure out where the documentation on the ADC that specifies the limitation is located, so I have a complete picture, and don't do anything...too stupid.. :) In the docs for Megatiny it mentions 3MHz for internal references and 6 MHz for external references, but I can find no mention of any limit in the data sheet, or ADC docs linked.  So I really feel like I am missing some documentation!  I have one counter example, which is certainly a nutty one but at https://ww1.microchip.com/downloads/en/Appnotes/TB3257-How-to-Use-the-12-Bit-Differential-ADC-in-Series-Accumulation-Mode-DS90003257A.pdf in the minimum sample time it gives an 8MHz clock.  (and then numbers that appear to go beyond the datasheet specs for max conversion rate).  The point being, either I'm missing some documentation, and I strongly believe that to be the case since the author(s?) of this library seem way more informed on these chips, or something else (Atmel changed documentation?  Limits apply to other chips?  Limits are posted in other chips and Atmel neglected to publish in the 3226 datasheet?  In fact, I'll look at the EA datasheet since it appears to use the same ADC, but even then, are we certain they're the same?)  

The entire project I'm using this for is feature creep....and I recognize that, and am still allowing for it, so I'm already starting out as a mistake, but if I'm going to feature creep, I want to creep to the maximum extent possible :)  So I want to start with as much information as I can get.

**Comment by clazarowitz** [comment]
I should also add, I'd just try stuff if I had parts/boards already :)  Right now I'm just trying to get code ready for when I do have the boards.  I looked to get a board off the tindie store too, just to have a dedicated playground, since I'm not even 100% sure my boards will work and may need some revisions, but it looks like they still have you frozen out or something.  It says in stock, but attempts to add it to the cart say out of stock.  Given your message on the store, I am guessing this is already a known issue :(

**Reply by SpenceKonde** [reply]
What? The boards should not be out of stock anymore, and people are buyinfg stuff from me normally, I'll see what's up with 3226.

**Reply by clazarowitz** [reply]
Weird....based on your response I tried another browser and apparently its browser incompatibilities.  I generally am lazy and do not install extra browsers on whichever platform I'm on (Windows-edge, Linux-firefox, MacOS/ios-safari, android-chrome) I run pretty much every OS various places!)  Anyhow, I've been finding more and more these days that despite many of them running the same browser engine (chrome), there are weird incompatibilities...this appears to be one of them.  With an alternate browser it let me order.  I picked options that will most closely match how I have my board setup.  I attached a screenshot of the apparent incompatibility.  Feel free to pass it on to tindie.
![NotCompatible](https://github.com/SpenceKonde/megaTinyCore/assets/87946568/7662b313-69c8-48eb-95fa-0d510105aa98)


I am still curious about the ADC Clock.  I have my code setup for 3,333,333 (CLK_PER/6) for the ADC clock right now (I think as it looks like the peripheral clock is being set to the main clock, which is fine with me!) which I think that's the same divisor your code is using for "3000". That honestly should be plenty fast enough based on some quick/naive/ideal inductor calculations.  But Murphy is always there to kill your assumptions (along with your hopes and dreams).  So, I figure having full knowledge of the limits of the ADC is the first step in the chain for countering Murphy, or put another way, making yourself miserable enough to pay Murphey up front.  He loves to charge late fees, so its always best to pay early. So, I'm looking to fully understand what I'm seeing even before seeing the chips/boards.  Same reason for buying a dev board that will probably be usable before my own integrated board, and same reason for buying 2...buy 1 and it will get dropped onto something metal and die, even when you believed you removed all metal from the area :)  (Murphey finds a way!)
